### PR TITLE
Feat: Addon rbac permission utilities

### PR DESCRIPTION
**Pull request by yue9944882**

### CODE REVIEW

File: pkg/utils/permission.go [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+			}
+		}
+		return nil
+	})
+	return p
+}
+
+func (p *permissionBuilder) WithStaticClusterRoleBinding(binding *rbacv1.ClusterRoleBinding) RBACPermissionBuilder {
+	p.u.fns = append(p.u.fns, func(cluster *clusterv1.ManagedCluster, addon *addonapiv1alpha1.ManagedClusterAddOn) error {
+		ensureAddonOwnerReference(&binding.ObjectMeta, addon)
```
Comment (qiujian16) [inline pkg/utils/permission.go]: We cannot add this since this is a clusterrole

File: pkg/utils/permission.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+					Kind: rbacv1.GroupKind,
+					Name: userGroup,
+				},
+			},
+		})
+}
+
+func (p *permissionBuilder) WithStaticClusterRole(clusterRole *rbacv1.ClusterRole) RBACPermissionBuilder {
+	p.u.fns = append(p.u.fns, func(cluster *clusterv1.ManagedCluster, addon *addonapiv1alpha1.ManagedClusterAddOn) error {
+		ensureAddonOwnerReference(&clusterRole.ObjectMeta, addon)
```
Comment (qiujian16) [inline pkg/utils/permission.go]: We cannot add this since this is a clusterrole

File: pkg/utils/permission.go [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+		exists := true
+		latest, err := p.kubeClient.RbacV1().RoleBindings(cluster.Name).Get(context.TODO(), binding.Name, metav1.GetOptions{})
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return err
+			}
+			exists = false
+		}
+		if !exists {
+			_, err := p.kubeClient.RbacV1().RoleBindings(cluster.Name).Create(context.TODO(), binding, metav1.CreateOptions{})
```
Comment (qiujian16) [inline pkg/utils/permission.go]: use ApplyRole in library-go? I think we need to handle the rule change on rbac

File: pkg/utils/permission.go (hunk 1 above)
Comment (yue9944882) [reply pkg/utils/permission.go]: removed ownerref

File: pkg/utils/permission.go (hunk 2 above)
Comment (yue9944882) [reply pkg/utils/permission.go]: removed ownerref

File: pkg/utils/permission.go (hunk 3 above)
Comment (yue9944882) [reply pkg/utils/permission.go]: good point, replace w `ApplyXXX`

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**